src/main/java/org/jenkinsci/plugins/periodicbackup/S3.java

```diff
@@ -0,0 +1,303 @@
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Logger;
+
+import org.acegisecurity.AccessDeniedException;
+import org.apache.commons.lang.StringUtils;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.interceptor.RequirePOST;
+
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import com.cloudbees.jenkins.plugins.awscredentials.AmazonWebServicesCredentials;
+import com.cloudbees.plugins.credentials.CredentialsMatchers;
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.google.common.base.Objects;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Sets;
+import com.google.common.io.Files;
+
+import hudson.Extension;
+import hudson.RestrictedSince;
+import hudson.security.ACL;
+import hudson.util.FormValidation;
+import hudson.util.IOUtils;
+import hudson.util.ListBoxModel;
+import jenkins.model.Jenkins;
+
+/**
+ *
+ * LocalDirectory defines the local folder to store the backup files
+ */
+public class S3 extends Location {
+
+	private String bucket;
+	private String tmpDir;
+	private String region;
+	private String credentialsId;
+
+	private static final Logger LOGGER = Logger.getLogger(S3.class.getName());
+
+	@DataBoundConstructor
+	public S3(String bucket, boolean enabled, String tmpDir, String region, String credentialsId) {
+		super(enabled);
+		this.bucket = bucket;
+		this.setTmpDir(tmpDir);
+		this.setRegion(region);
+		this.setCredentialsId(credentialsId);
+	}
+
+	@SuppressWarnings("deprecation")
+	@Override
+	public Iterable<BackupObject> getAvailableBackups() {
+		AmazonS3 client = AmazonUtil.getAmazonS3Client(region, credentialsId);
+
+		List<S3ObjectSummary> objectSummarys = client.listObjects(bucket).getObjectSummaries();
+		List<String> backupObjectFileNames = new ArrayList<String>();
+		List<File> backupObjectFiles = new ArrayList<File>();
+		for (S3ObjectSummary objectSummary : objectSummarys) {
+			if (StringUtils.endsWith(objectSummary.getKey(), BackupObject.EXTENSION)) {
+				backupObjectFileNames.add(objectSummary.getKey());
+				try {
+					File dir = new File(tmpDir);
+					if (!dir.isDirectory()) {
+						if (!dir.mkdir()) {
+							LOGGER.warning("Unable to make temp directory: " + tmpDir);
+							return null;
+						}
+					}
+					File file = new File(dir + objectSummary.getKey());
+					IOUtils.copy(client.getObject(bucket, objectSummary.getKey()).getObjectContent(),
+							new FileOutputStream(file));
+					backupObjectFiles.add(file);
+				} catch (Exception e) {
+					// TODO Auto-generated catch block
+					e.printStackTrace();
```

Comment:
We should use a logger here instead of stdout.

Reply:
Done